punchbowl/levelq/pca.py

```diff
@@ -11,114 +18,203 @@
-                index_to_insert += 1
-                bodies_in_quarter.append(find_bodies_in_image(input_file))
+            data, meta = input_file.data, input_file.meta
+            body_finding_input = (input_file.meta, input_file.wcs)
+            uncertainty_is_inf = np.isinf(input_file.uncertainty.array)
         elif isinstance(input_file, str):
-            cube = load_image_task(input_file, include_provenance=False, include_uncertainty=False)
-            if check_file(cube.meta["DATAAVG"].value, cube.meta["DATAMDN"].value):
-                all_files_to_fit[index_to_insert] = cube.data
-                index_to_insert += 1
-                bodies_in_quarter.append(find_bodies_in_image(cube))
-        elif isinstance(input_file, Callable):
-            mean, median, bodies = input_file(all_files_to_fit[index_to_insert])
-            if check_file(mean, median):
-                # This is a bad file. Don't increment the insertion index, so this one gets overwritten by the next file
-                index_to_insert += 1
-                bodies_in_quarter.append(bodies)
+            cube = load_image_task(input_file, include_provenance=False, include_uncertainty=True)
+            data, meta = cube.data, cube.meta
+            body_finding_input = (cube.meta, cube.wcs)
+            uncertainty_is_inf = np.isinf(cube.uncertainty.array)
+        elif isinstance(input_file, DataLoader):
+            data, meta, wcs, uncertainty_is_inf = input_file.load()
+            body_finding_input = (meta, wcs)
         else:
             raise TypeError(f"Invalid type {type(input_file)} for input file")
+        if check_file(meta, outlier_limits):
+            loaded_input_list_indices.append(subtract)
+            all_files_to_fit[index_to_insert] = data
+            index_to_insert += 1
+            body_finding_inputs.append(body_finding_input)
+            is_masked *= uncertainty_is_inf * (data == 0)
+        else:
+            n_outliers += 1
+
+    loaded_input_list_indices = np.array(loaded_input_list_indices)
 
     # Crop the unused end of the array
     all_files_to_fit = all_files_to_fit[:index_to_insert]
-    logger.info(
-        f"Loaded {len(files_to_fit)} images to fit")
-    logger.info(f"Kept {index_to_insert}, filling {all_files_to_fit.nbytes / 1024 ** 3:.2f} GB")
+    logger.info(f"Total of {len(all_files_to_fit)} images to fit, filling {all_files_to_fit.nbytes / 1024 ** 3:.2f} GB")
+    logger.info(f"(Drawn from {len(input_cubes)} images to subtract and {len(files_to_fit)} extra images for fitting)")
+    logger.info(f"({n_outliers} outliers were rejected from fitting)")
 
-    bodies_in_quarter = np.array(bodies_in_quarter)
+    logger.info("Locating planets")
+    # We have a lot of data in memory right now, so forking is expensive as all that memory has to be marked as
+    # copy-on-write. Using a forkserver avoids that work.
+    ctx = mp.get_context("forkserver")
```

Comment:
This definitively means we don't support Windows, right? It's only supported on POSIX systems.